quickwit/quickwit-search/src/collector.rs

```diff
@@ -1088,32 +880,20 @@ impl Collector for QuickwitCollector {
         };
         let score_extractor = get_score_extractor(&self.sort_by, segment_reader)?;
         let (order1, order2) = self.sort_by.sort_orders();
-        let sort_key_mapper = HitSortingMapper { order1, order2 };
-        // Precompute the order for search_after if split_id is set
-        let precomp_search_after_order = match &self.search_after {
-            Some(search_after) if !search_after.split_id.is_empty() => order1
-                .compare(&self.split_id, &search_after.split_id)
-                .then_with(|| order1.compare(&segment_ord, &search_after.segment_ord)),
-            // This value isn't actually used.
-            _ => Ordering::Equal,
-        };
-        // Convert search_after into fast field u64
-        let search_after =
-            SearchAfterSegment::new(self.search_after.clone(), order1, order2, &score_extractor);
 
         let segment_top_k_collector = if leaf_max_hits == 0 {
```

Comment:
Arguable: now that you have dynamic dispatch, the code might be a tiny bit cleaner if we removed the `Option<Box<dyn QuickwitSegmentTopKCollector>>` and just made it a `Box<dyn QuickwitSegmentTopKCollector>`.

The logic `if leaf_max_hits == 0` would then be moved in `specialized_top_k_segment_collector`
and we would return a `DoNothingSegmentTopKCollector`.

(I wouldn't expect any performance difference, one way or another. It is just to remove the Option)

Reply:
I think it's a little bit more readable to have an option instead of a `DoNothingCollector` behind the trait